Source/ACE/Entity/Landblock.cs

```diff
@@ -186,6 +188,39 @@ public void RemoveWorldObject(ObjectGuid objectId, bool adjacencyMove)
             }
         }
 
+        /// <summary>
+        /// Handle the QueryHealth action between the source Object and its target
+        /// </summary>
+        public void HandleQueryHealth(Session source, ObjectGuid targetId)
+        {
+            if (targetId.IsPlayer())
+            {
+                Player pl = null;
+
+                lock (objectCacheLocker)
+                {
+                    if (this.worldObjects.ContainsKey(targetId))
+                        pl = (Player)this.worldObjects[targetId];
+                }
+                if (pl == null)
+                {
+                    // check adjacent landblocks for the targetId
+                    foreach (var block in adjacencies)
+                    {
+                        lock (objectCacheLocker)

```

Comment:
Good thinking, but unfortunately this particular lock isn't going to help you.  You're locking THIS landblock's mutex and poking your head into other landblocks.  You need to lock block.objectCacheLocker.